### PR TITLE
Fix i18n

### PR DESCRIPTION
**Pull request by toitzi**
# Details

This PR adds an Error_Exception Class which extends ErrorException (with additional support of translation).

It also makes sure translation function` __()` is only used in APPATH. In MODPATH and SYSPATH `I18n::get()` function is used.

### Related Issue

Fixes Issue #293 completley.
Also fixes PR #292 

### How Has This Been Tested

Tested with unittests.

### Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by toitzi** [comment]
Please Review

File: system/classes/KO7/I18n.php [hunk 1]
```diff
@@ -65,75 +74,100 @@ public static function lang($lang = NULL)
...
+			$string = $table[$string] ?? $string;
+		}
+
+		return empty($values) ? $string : strtr($string, $values);
 	}
 
 	/**
 	 * Returns the translation table for a given language.
 	 *
-	 *     // Get all defined Spanish messages
```
Comment (jstrobel) [inline system/classes/KO7/I18n.php]: Spanish? I think there is a mixure, 2 lines later en-us

File: system/tests/ko7/I18nTest.php [hunk 2]
```diff
@@ -9,9 +8,10 @@
  *
  * @package    KO7
  * @category   Tests
- * @author     Kohana Team
- * @author     Jeremy Bush <contractfrombelow@gmail.com>
- * @copyright  (c) Kohana Team
+ *
+ * @author     Kohana Team, Jeremy Bush <contractfrombelow@gmail.com>
+ * @copyright  (c) 2008´- 2016 Kohana Team
```
Comment (jstrobel) [inline system/tests/ko7/I18nTest.php]: 2008´? Think it is 2008 - 2016 or did i get something wrong? :)

File: system/classes/KO7/I18n.php (hunk 1 above)
Comment (toitzi) [reply system/classes/KO7/I18n.php]: Thx :)

File: system/tests/ko7/I18nTest.php (hunk 2 above)
Comment (toitzi) [reply system/tests/ko7/I18nTest.php]: Yes ofc ;) thanks for the details 😄

**Pull request merged**